### ISSUE TITLE
clinicalEnrichment & plotEnrichmentResults - Mixed Reference Groups 

### ISSUE THREAD

**Issue by PoisonAlien** [PoisonAlien/maftools#715]
Heylo!

This seemed more like a question, rather than an issue as such, but let me know if it would have been better in 'Issues'.

When using clinicalEnrichment and plotEnrichmentResults, we obtain a plot as below;

![Example](https://user-images.githubusercontent.com/75019173/118653205-3170d400-b7df-11eb-8612-565195b5ec09.png)

We see the '0' group used as group 1, then suddenly in 9th gene swaps and uses '1' as group 2. The relationships is the same, '0' having a lesser % mutated. As far as I can tell, the p value, which is used to rank the genes, is the same whichever way round you have the groups.

I'm wondering, how is it then decided which way it is plotted? For example, we get the same information (I think) if the 9th gene used '0' group as group 1, and this would be easier to interpret visually as we are not flipping the groups around. 

Would it be clearer to interpret if we uniformly used '0' or '1' as group 1? This would also make it easier to instantly see which genes are relatively enriched in a positive or negative manner. Additionally, if the 'rest' bar was also coloured by its group, this might also make it clearer visually?

Is there a purpose behind the switch? I think it might be quite jarring for an audience who will instinctively be looking for a reason why the groups are switched.

I fear I may be missing something obvious :)

Thanks!

_Originally posted by @HHayman in https://github.com/PoisonAlien/maftools/discussions/713_

**Comment by PoisonAlien** [comment]
Hi,

I have added `ORthr` and `featureLvls` argument to `plotEnrichmentResults`. This should enable you to limit platting to selected groups or filtering based on odds ratios. 
Let me know if this helps.

**Comment by HHayman** [comment]
Apologies, trying to update MafTools is causing big problems, so I can't test it at the moment.

I don't think the problem is MafTools though but rather Rtools. Once I resolve that issue I will get back to you, sorry!

**Comment by PoisonAlien** [comment]
Sure no problem. BTW, the new rtools4 has been released today.  Read [here](https://cran.r-project.org/bin/windows/Rtools/) which might help.

**Comment by HHayman** [comment]
I've been unable to resolve the update issue, but I seem to pass all of the checks from the installation instructions, so I just want to check if there is anything MafTools specific that might throw up this issue so that I can be sure that it isn't related to MafTools;

```r
* installing *source* package 'maftools' ...
** using staged installation
** libs

*** arch - i386
"C:/rtools40/mingw32/bin/"gcc  -I"C:/PROGRA~1/R/R-41~1.0/include" -DNDEBUG -D_FILE_OFFSET_BITS=64 -I'C:/Program Files/R/R-4.1.0/library/Rhtslib/include' -I'C:/Program Files/R/R-4.1.0/library/zlibbioc/include'        -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c ntcounts.c -o ntcounts.o
"C:/rtools40/mingw32/bin/"gcc  -I"C:/PROGRA~1/R/R-41~1.0/include" -DNDEBUG -D_FILE_OFFSET_BITS=64 -I'C:/Program Files/R/R-4.1.0/library/Rhtslib/include' -I'C:/Program Files/R/R-4.1.0/library/zlibbioc/include'        -O2 -Wall  -std=gnu99 -mfpmath=sse -msse2 -mstackrealign  -c somaticfreq.c -o somaticfreq.o
C:/rtools40/mingw32/bin/gcc -shared -s -static-libgcc -o maftools.dll tmp.def ntcounts.o somaticfreq.o C:/Program Files/R/R-4.1.0/library/Rhtslib/usrlib/i386/libhts.a -L/lib/i386 -lcurl -lrtmp -lssl -lssh2 -lcrypto -lgdi32 -lz -lws2_32 -lwldap32 -lwinmm -lidn -LC:/PROGRA~1/R/R-41~1.0/bin/i386 -lR
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lcurl
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lrtmp
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lssl
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lssh2
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lcrypto
C:/rtools40/mingw32/bin/../lib/gcc/i686-w64-mingw32/8.3.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot find -lidn
collect2.exe: error: ld returned 1 exit status
no DLL was created
ERROR: compilation failed for package 'maftools'
* removing 'C:/Program Files/R/R-4.1.0/library/maftools'
* restoring previous 'C:/Program Files/R/R-4.1.0/library/maftools'
* ```

**Comment by ShixiangWang** [comment]
@HHayman Are you using an i386 (32-bit) machine/gcc? Did you specify a wrong rtools path?

**Comment by HHayman** [comment]
It's a 64-bit machine and as far as I can tell, path is correct. Strange.

**Comment by PoisonAlien** [comment]
I don't have access to a Windows machine. I will try my best and let you know what I can do..

**Comment by HHayman** [comment]
Thank you very much :)

**Comment by HHayman** [comment]
On the topic of plotEnrichmentResults, is it a simple change to apply the colouring by variant classification to this output, as with coOncoplot and coBarplot?

**Comment by PoisonAlien** [comment]
Yes, it should be simple. You can pass a named vector of color codes for every level in the clinical feature. You can check the levels in the results under the column `cf`
e.g;
 `res$cf_sizes$cf`

**Comment by HHayman** [comment]
Sorry, I think I described that badly. That would allow you to choose a colour scheme for the levels of your clinical feature if I understand? I mean in the way that missense etc are coloured automatically when running coOncoplot or coBarplot, as below;


![Rplot](https://user-images.githubusercontent.com/75019173/119496100-0d6f3e80-bd5b-11eb-9fd0-63d6e86d9e93.jpeg)

**Comment by PoisonAlien** [comment]
So if I understand correctly, you would like to have the bars in the `plotEnrichmentResults` as stacked barplot of variant classifications?

**Comment by HHayman** [comment]
Exactly :)

**Comment by PoisonAlien** [comment]
Hi,
Turns out it's a bit more complicated than I thought. It certainly is doable but would require quite a bit of code change. 
If you really would like to show the variant classifications for these genes, you could just draw the `oncoplot` with the clinical feature as an annotation. Or subset your MAF for your feature of interest and draw `mafbarplot` .

**Comment by HHayman** [comment]
I've avoided Oncoplots lately because I was having spatial problems that I couldn't resolve, but I'll play around with things and see if I can achieve it this way. Thanks :)
